src/sparseml/pytorch/sparsification/pruning/mask_creator.py

```diff
@@ -130,9 +130,24 @@ def create_sparsity_masks(
                 masks.append(tensor.new_ones(tensor.shape))
                 continue
 
+            num_elem = tensor.numel()
+            target_num_mask = round(num_elem * sparsity_target)
             min_val = tensor.min().item()
+
             if threshold.item() > min_val:
-                masks.append((tensor > threshold).type(tensor.type()))
+                threshold_mask = (tensor > threshold).type(tensor.type())
```

Comment:
A one-off error occurs here, but it seems it stems from `_threshold_from_sparsity()` and not from the changes here. If threshold is meant to be the value s.t. all tensors > threshold are kept, then line 189 should be changed to:
`round(sparsity * tensor.numel()) - 1`